Panic during start period

Sometimes I was able to observe a following panic:

```
time="2020-06-26T18:01:16Z" level=info msg="Got leader lease"
time="2020-06-26T18:01:16Z" level=info msg="Waiting for initial cache sync"
time="2020-06-26T18:01:16Z" level=info msg="cache synced"
E0626 18:01:19.272552       1 runtime.go:73] Observed a panic: "invalid memory address or nil pointer dereference" (runtime error: invalid memory address or nil pointer dereference)
goroutine 88 [running]:
k8s.io/apimachinery/pkg/util/runtime.logPanic(0x106f560, 0x235a530)
	/go/pkg/mod/k8s.io/apimachinery@v0.0.0-20190612205821-1799e75a0719/pkg/util/runtime/runtime.go:69 +0x58
k8s.io/apimachinery/pkg/util/runtime.HandleCrash(0x0, 0x0, 0x0)
	/go/pkg/mod/k8s.io/apimachinery@v0.0.0-20190612205821-1799e75a0719/pkg/util/runtime/runtime.go:51 +0x7c
panic(0x106f560, 0x235a530)
	/usr/local/go/src/runtime/panic.go:522 +0x1a0
github.com/wish/nodereaper/pkg/deletion.(*Deleter).killMyselfFirst(0x400022a140, 0x40005882c0)
	/go/src/github.com/wish/nodereaper/pkg/deletion/deletion.go:203 +0xe8
github.com/wish/nodereaper/pkg/deletion.(*Deleter).pollDeletions(0x400022a140)
	/go/src/github.com/wish/nodereaper/pkg/deletion/deletion.go:167 +0xfe0
github.com/wish/nodereaper/pkg/deletion.(*Deleter).Run.func1()
	/go/src/github.com/wish/nodereaper/pkg/deletion/deletion.go:68 +0x4c
k8s.io/apimachinery/pkg/util/wait.JitterUntil.func1(0x40008a0fa0)
	/go/pkg/mod/k8s.io/apimachinery@v0.0.0-20190612205821-1799e75a0719/pkg/util/wait/wait.go:152 +0x50
k8s.io/apimachinery/pkg/util/wait.JitterUntil(0x40008a0fa0, 0x37e11d600, 0x0, 0x1543c01, 0x4000090660)
	/go/pkg/mod/k8s.io/apimachinery@v0.0.0-20190612205821-1799e75a0719/pkg/util/wait/wait.go:153 +0xb8
k8s.io/apimachinery/pkg/util/wait.Until(0x40008a0fa0, 0x37e11d600, 0x4000090660)
	/go/pkg/mod/k8s.io/apimachinery@v0.0.0-20190612205821-1799e75a0719/pkg/util/wait/wait.go:88 +0x48
created by github.com/wish/nodereaper/pkg/deletion.(*Deleter).Run
	/go/src/github.com/wish/nodereaper/pkg/deletion/deletion.go:66 +0xa4
panic: runtime error: invalid memory address or nil pointer dereference [recovered]
	panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x40 pc=0xe7d288]

goroutine 88 [running]:
k8s.io/apimachinery/pkg/util/runtime.HandleCrash(0x0, 0x0, 0x0)
	/go/pkg/mod/k8s.io/apimachinery@v0.0.0-20190612205821-1799e75a0719/pkg/util/runtime/runtime.go:58 +0x104
panic(0x106f560, 0x235a530)
	/usr/local/go/src/runtime/panic.go:522 +0x1a0
github.com/wish/nodereaper/pkg/deletion.(*Deleter).killMyselfFirst(0x400022a140, 0x40005882c0)
	/go/src/github.com/wish/nodereaper/pkg/deletion/deletion.go:203 +0xe8
github.com/wish/nodereaper/pkg/deletion.(*Deleter).pollDeletions(0x400022a140)
	/go/src/github.com/wish/nodereaper/pkg/deletion/deletion.go:167 +0xfe0
github.com/wish/nodereaper/pkg/deletion.(*Deleter).Run.func1()
	/go/src/github.com/wish/nodereaper/pkg/deletion/deletion.go:68 +0x4c
k8s.io/apimachinery/pkg/util/wait.JitterUntil.func1(0x40008a0fa0)
	/go/pkg/mod/k8s.io/apimachinery@v0.0.0-20190612205821-1799e75a0719/pkg/util/wait/wait.go:152 +0x50
k8s.io/apimachinery/pkg/util/wait.JitterUntil(0x40008a0fa0, 0x37e11d600, 0x0, 0x1543c01, 0x4000090660)
	/go/pkg/mod/k8s.io/apimachinery@v0.0.0-20190612205821-1799e75a0719/pkg/util/wait/wait.go:153 +0xb8
k8s.io/apimachinery/pkg/util/wait.Until(0x40008a0fa0, 0x37e11d600, 0x4000090660)
	/go/pkg/mod/k8s.io/apimachinery@v0.0.0-20190612205821-1799e75a0719/pkg/util/wait/wait.go:88 +0x48
created by github.com/wish/nodereaper/pkg/deletion.(*Deleter).Run
	/go/src/github.com/wish/nodereaper/pkg/deletion/deletion.go:66 +0xa4
```

--- a/pkg/deletion/deletion.go
+++ b/pkg/deletion/deletion.go
@@ -199,8 +199,14 @@ func (d *Deleter) killMyselfFirst() bool {
 	if err != nil || myNode == nil {
 		return true
 	}
+	groupKey := d.nodeGroupKey(myNode)
+	//sometimes the k8s api does not return own node in list node right on creation so the map might have not been populated yet.
+	if d.states.Groups[groupKey] == nil || d.states.Groups[groupKey].Nodes[myNode.Name] == nil{
+		logrus.Infof("Own node %v not found, skipping... ",d.opts.NodeName)
+		return false
+	}
 	// Keep going if we're already deleting
-	if d.states.Groups[d.nodeGroupKey(myNode)].Nodes[myNode.Name].State != DontWantDelete {
+	if d.states.Groups[groupKey].Nodes[myNode.Name].State != DontWantDelete {
 		return true
 	}
 	// Don't delete if we wouldn't want to

